### PR TITLE
Rafabios feature/ingress-and-svc-type

### PR DESCRIPTION
**Pull request by rafabios**
Added ingress templating suppot: is possible to use nginx, kong, traefik and so on as a ingress controller;

Added service type, so users can choose between ClusterIP, NodePort and LoadBalancer;

### CODE REVIEW

File: helm/sloop/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  {{- range $key, $value := .Values.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.name }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+  name: {{ .Values.name }}
+  #namespace: {{ .Values.namespace }}
```
Comment (mallow111) [inline helm/sloop/templates/ingress.yaml:15]: either remove this comment line or add namespace variables in Values file

File: helm/sloop/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  rules:
+  - host: {{ .Values.ingress.host }}
+    http:
+      paths:
+      - backend:
+          serviceName: {{ .Values.name }}
+          servicePort: {{ .Values.service.port }}
+        path: /
+{{- end -}}
+  {{- if .Values.ingress.tls }}
```
Comment (mallow111) [inline helm/sloop/templates/ingress.yaml:26]: just curious, ingress.tls is a {}, how does this work in if condition?

File: helm/sloop/templates/ingress.yaml (hunk 1 above)
Comment (rafabios) [reply helm/sloop/templates/ingress.yaml]: Done! removed namespace. Sorry about that!

File: helm/sloop/templates/ingress.yaml (hunk 2 above)
Comment (rafabios) [reply helm/sloop/templates/ingress.yaml:25]: if you remove {} it should accept kubernetes ingress format, like this:

```
 tls:
  - secretName: sloop-tls
    hosts:
     - sloop.example.com
```

Also a examplo from kubernetes-dashboard:
https://github.com/helm/charts/blob/7e45e678e39b88590fe877f159516f85f3fd3f38/stable/kubernetes-dashboard/values.yaml#L149

**Review by sana-jawad — APPROVED** [review]

**Pull request merged**